packages/smart-vaults/src/signers/MultiSigner.sol

```diff
@@ -56,16 +56,31 @@ library MultiSignerLib {
     /*                                  FUNCTIONS                                 */
     /* -------------------------------------------------------------------------- */
 
+    /// @notice Get signer at index from storage.
+    function getSigner(MultiSigner storage $_, uint8 index_) internal view returns (Signer memory) {
+        return $_.signers[index_];
+    }
+
+    /// @notice Get threshold from storage.
+    function getThreshold(MultiSigner storage $_) internal view returns (uint8) {
+        return $_.threshold;
+    }
+
+    /// @notice Get number of signers from storage.
+    function getSignerCount(MultiSigner storage $_) internal view returns (uint8) {
+        return $_.signerCount;
+    }
+
     /**
-     * @notice Sets signer at index in storage.
+     * @notice Adds signer at index in storage.
      *
      * @dev Throws error when signer is not EOA or Passkey.
      *
-     * @param $_ multi signer storage reference.
+     * @param $_ Multi signer storage reference.
      * @param signer_ Signer to be set.
-     * @param index_ Index to set signer at.
+     * @param index_ Index to set signer at. Assumed to be currently unoccupied.
```

Comment:
are we sure this shouldn't include the check itself? (instead of having in the other place?) at a minimum i'd put it up in `@dev` for more attention since it would be a ~big mistake to overwrite an existing signer and bump up the signer count. might want to call out the signer count then in `@dev` too

Reply:
Yes, for me the ideal would be to include the check itself here. Another reason is that overwriting a Passkey signer with an EOA/AccountSigner would leave the storage corrupted because slot2 is not zeroed.

The downside of including the check is it will be slightly more expensive (~300 gas?) when it's not needed, e.g. during initialization.